examples/elixir/udp_hole_punching/01-puncher.exs

```diff
@@ -0,0 +1,24 @@
+["setup.exs", "puncher.exs"] |> Enum.map(&Code.require_file/1)
+
+[my_name, their_name, addr] = System.argv()
+[rendezvous_host, port_s] = String.split(addr, ":")
+{rendezvous_port, ""} = Integer.parse(port_s)
+
+rendezvous_address = Ockam.Transport.UDPAddress.new(rendezvous_host, rendezvous_port)
+
+{:ok, _name} =
+  Puncher.create(
+    address: my_name,
+    attributes: %{
+      target: their_name,
+      rendezvous_address: rendezvous_address
+    }
+  )
+
+# Port 0 will cause the OS to assign a random port
+{:ok, _udp} = Ockam.Transport.UDP.start(port: 0)
+
+Ockam.Node.whereis(my_name)
+|> GenServer.call(:ping_rendezvous_server)
+
```

Comment:
I'd have expected here to send a msg to the other end,  and wait till receiving a message from the other end as well, to show how user code can make use of the  "Puncher" to send messages traversing NAT.  Not necessary like this,  but this is what would come to my mind:
```
# Register this process as worker address "app".  
Ockam.Node.register_address("app")

{:ok, puncher_addr} = Puncher.create(...)

# Send a msg to the puncher worker.   It will forward it to the other end
# and from there it will be routed to the "app" address on the other' end node..
Ockam.Router.route(%{onward_route: [puncher_addr, "app"],    message: "hello", ..})

# Since on this example there is no "client" and "server" but both are 01-puncher.exs instances,  the peer will send
# a msg to us as well (the previous command).  Receive it and print it.
receive do
   msg -> print the msg
end

```

Reply:
I'm glad you brought this one up, I saw that self registration and receive block in other examples, but in practice I've never gone to production with code containing an explicit receive block. As such I just made the worker module (Puncher in this case) handle receiving since its already wired to do so. But as the other examples are more along the lines as yours, I'm curious if that is providing additional benefit that I'm not seeing.